HHH.playground/Contents.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+
+class Spoon {
+    private let lock = NSLock()
+    var index: Int
+     
+    func pickUp() {
+        lock.lock()
+    }
+    
+    func putDown() {
+        lock.unlock()
+    }
+    
+    init(index: Int) {
+        self.index = index
+    }
+}
+
+class Developer {
+    
+    var leftSpoon: Spoon
+    var rightSpoon: Spoon
+    var randomTime = UInt32.random(in: 5...10)
+    var name: String
+    
+    func think() {
+        if rightSpoon.index > leftSpoon.index {
+            leftSpoon.pickUp()
+        } else {
+            rightSpoon.pickUp()
+        }
```

Comment:
You were supposed to have them pick up both spoons before it returns like this...
```suggestion
        if rightSpoon.index > leftSpoon.index {
            leftSpoon.pickUp()
            rightSpoon.pickUp()
        } else {
            rightSpoon.pickUp()
            leftSpoon.pickUp()
        }
```
your code still works just fine, I tested it but just thought I would let you know. Great job on writing concurrent code Ron!